module3/homework/passwordCheck/validation.cpp

```diff
@@ -1,2 +1,86 @@
 #include "validation.hpp"
-// TODO: Put implementations here
\ No newline at end of file
+#include <string>
+#include <iostream>
+#include <algorithm>
+
+bool czy_zawiera_cyfre(char znak)
+{
+    int cyfra = (int)znak;
+    return (cyfra >= 48) && (cyfra <= 57);
+}
+
+bool czy_zawiera_znak_specjalny(char znak)
+{
+    int cyfra = (int)znak;
+    return ((cyfra >= 0) && (cyfra <= 47)
+        || (cyfra >= 58) && (cyfra <= 64)
+        || (cyfra >= 91) && (cyfra <= 96)
+        || (cyfra >= 0) && (cyfra <= 47)
+        || (cyfra >= 123) && (cyfra <= 255));
+}
+
```

Comment:
don't use  "magic numbers"
```suggestion
    return ((cyfra >= 0) && (cyfra <= 47)
        || (cyfra >= 58) && (cyfra <= 64)
        || (cyfra >= 91) && (cyfra <= 96)
        || (cyfra >= 0) && (cyfra <= 47)
        || (cyfra >= 123) && (cyfra <= 255));
}

```